### PR TITLE
fix(linter): bug in fixer for prefer-to-have-length

### LINKED ISSUE #5132: bug(linter): bug in fixer for `prefer-to-have-length`
split from https://github.com/oxc-project/oxc/issues/5103

```
bunx oxlint@latest --jest-plugin --jsdoc-plugin --jsx-a11y-plugin --nextjs-plugin --react-perf-plugin -D prefer-to-have-length --fix <path name>
```

input

```ts
expect((meta.get('pages') as YArray<unknown>).length).toBe(
  (originalMeta.get('pages') as YArray<unknown>).length
);
```

output:
```ts
 (originalMeta.get('pages') as YArray<unknown>).length
);
```
expected output
```ts
expect((meta.get('pages') as YArray<unknown>)).toHaveLength(
  (originalMeta.get('pages') as YArray<unknown>).length
);
```

### PR DESCRIPTION
**Pull request by shulaoda**
closes #5132 

I exposed the `new_fix` of `RuleFixer` to support custom multi-step fixes, and I'm not sure if this is allowed.

### CODE REVIEW

**Comment by shulaoda** [comment]
There may be a simpler way to fix this, just wait a moment.

File: crates/oxc_linter/src/rules/jest/prefer_to_have_length.rs [hunk 1]
```diff
@@ -138,52 +118,30 @@ impl PreferToHaveLength {
...
+                        static_mem_expr.span().end,
+                    )),
+                    Fix::new("toHaveLength", matcher_span),
+                ]),
+                Some(Cow::Owned(format!(
+                    "Use `toHaveLength` instead of `{}`.",
+                    matcher.name().unwrap()
+                ))),
+            )
+        });
```
Comment (camc314) [inline crates/oxc_linter/src/rules/jest/prefer_to_have_length.rs:144]: you should just be able to return a iterator (then colected into `RuleFix`

https://github.com/oxc-project/oxc/blob/34bfaf67922b5217584940ad49c1d363437fa9ee/crates/oxc_linter/src/rules/react/jsx_props_no_spread_multi.rs#L97-L104

**Comment by shulaoda** [comment]
Two commits represent two different repair methods. Which do you think is better? @camc314

**Review by camc314 — APPROVED** [review]
this way looks good to me

**Review by DonIsaac — APPROVED** [review]

**Pull request merged**